client/js/router.js

```diff
@@ -94,9 +94,57 @@ router.beforeEach((to, from, next) => {
 		return;
 	}
 
-	// Disallow navigating to invalid channels
-	if (to.name === "RoutedChat" && !store.getters.findChannel(Number(to.params.id))) {
-		next(false);
+	// If trying to navigate to an invalid channel,
+	// we attempt to either open a connection dialog to the network
+	// or populate the Join Channel field in the existing network.
+	if (to.name === "RoutedChat") {
+		let channel = to.hash;
+		const {networkHost, channelName} = to.params;
+
+		// If the channel isn't provided as the hash, check if it's provided as the next param
+		if (!channel) {
+			if (channelName) {
+				channel = channelName;
+			}
+		}
+
+		if (store.getters.findChannelByName(networkHost, channel)) {
+			next();
+			return;
+		}
+
+		const existingNetwork = store.state.networks.find(
+			(network) => network.host === to.params.networkHost
+		);
+
+		if (existingNetwork) {
+			// Join Channel UI
+
+			const activeChannel = store.state.activeChannel;
+
+			// if the active channel is in the network, send the user back to that channel, else to the lobby
+			if (activeChannel && activeChannel.network.uuid === existingNetwork.uuid) {
+				next({
+					path: `/${to.params.networkHost}/${encodeURIComponent(
```

Comment:
This seems like the only place you called `encodeURIComponent`. If I had to guess there will be problems with unconventional channel names (like containing `/` and other fancy characters that may break the url).